### PR TITLE
Added Case Studies page in Overview Section 

### PR DESCRIPTION
**Pull request by TamannaVerma99**
**What kind of change does this PR introduce?**
Added a new page in Overview section to show the case studies published in the blog. Currently, since I don't have access to the data file that has information related to blogs published, I have created a sample file of my own named `casestudies.json` with the same data format as mentioned in the issue and linked it to the card component. This makes the JSON file easily replaceable with the type of data we'll be using.

**Issue Number:**

-  Related to #440  

**Does this PR introduce a breaking change?**
No

### CODE REVIEW

File: components/Sidebar.tsx [hunk 1]
```diff
@@ -284,6 +286,7 @@ export const DocsNav = () => {
             label='Similar Technologies'
           />
           <DocLink uri='/overview/code-of-conduct' label='Code of Conduct' />
+          <DocLink uri='/overview/casestudies' label='Casestudies' />
```
Comment (benjagm) [inline components/Sidebar.tsx:289]:
```suggestion
          <DocLink uri='/overview/casestudies' label='Case Studies' />
```

File: pages/overview/casestudies.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
```
Comment (benjagm) [inline pages/overview/casestudies.md:1]: Instead of md file i think it will be better to create a tsx file this way: pages/overview/casestudies/index.page.tsx.

See an example of this type of page in the sponsors page inside overview. You can have the title and intro paragraph in a md file and later put the grid code in the tsx page. See implementations page for inspiration.

File: components/CustomComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
```
Comment (benjagm) [inline components/CustomComponent.tsx:1]: Instead of a component for the whole list, we will be using a standard component created in this PR:https://github.com/json-schema-org/website/issues/433

I think you are not going to need to create a component, instead use the one created by Michael-Obele in #433.

The logic to build the grid is something you will add in pages/overview/casestudies/index.page.tsx.

**Review by benjagm — CHANGES_REQUESTED** [review]
This a really great first draft TamannaVerma99 !!!

I am leaving some comments to reorganize the solution but great job so far. Remember we'll use the card component that is being in process in #433

**Comment by TamannaVerma99** [comment]
Thanks @benjagm !
I just want to reconfirm are we going to use standardized cards for this page too?As in the issue you mentioned we are not going to use standardized cards we are instead going to use them for welcome page.
I am working on other feedbacks you have given.Thanks :)

**Comment by benjagm** [comment]
> Thanks @benjagm ! I just want to reconfirm are we going to use standardized cards for this page too?As in the issue you mentioned we are not going to use standardized cards we are instead going to use them for welcome page. I am working on other feedbacks you have given.Thanks :)

Thanks! In the initial requirements of this issue we did not speak about common card component, however I did it in this other comment of that issue. I realized that 3 people were working in a very similar problem so is better to standarize and reuse.

https://github.com/json-schema-org/website/issues/433#issuecomment-1975346307

**Comment by benjagm** [comment]
Can you please change the target branch of this PR to : https://github.com/json-schema-org/website/tree/web-release-3

Reference docs: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request

File: data/casestudies.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
+[
```
Comment (benjagm) [inline data/casestudies.json:1]: Can you please build a real data set with our case studies? This is the list:
https://json-schema.org/blog?type=Case%20Study

File: components/Sidebar.tsx [hunk 5]
```diff
@@ -183,7 +184,7 @@ export const SidebarLayout = ({ children }: { children: React.ReactNode }) => {
           </div>
         </div>
 
-        <div className='max-w-[1400px] mx-auto grid grid-cols-1 lg:grid-cols-4 mx-4 md:mx-12'>
+        <div className='max-w-[1400px] grid grid-cols-1 lg:grid-cols-4 mx-4 md:mx-12'>
```
Comment (benjagm) [inline components/Sidebar.tsx:187]: Why this change?

File: components/Layout.tsx [hunk 6]
```diff
@@ -164,7 +164,12 @@ const MainNavigation = () => {
         isActive={section === 'community'}
       />
       <div className='flex max-sm:ml-4 items-center gap-6 md:gap-4'>
-        <div className='flex justify-center rounded border-2 border-gray-100 ml-0 w-[120px] md:w-full'>
+        <div
```
Comment (benjagm) [inline components/Layout.tsx:167]: You don't need to edit this page.

File: components/Card.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
```
Comment (benjagm) [inline components/Card.tsx:1]: Please use the Card component available in the branch and modify it if necessary.

File: pages/overview/casestudies/_index.md [hunk 8]
```diff
@@ -0,0 +1,4 @@
+../../../_includes/community/programs/casestudies/casestudies.md
```
Comment (benjagm) [inline pages/overview/casestudies/_index.md:1]: Please remove this md file. Let's put all the content in the tsx page.

File: pages/overview/casestudies/index.page.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+import { getLayout } from '~/components/Sidebar';
+import fs from 'fs';
+import Head from 'next/head';
+import { Headline1 } from '~/components/Headlines';
+import matter from 'gray-matter';
+import StyledMarkdown from '~/components/StyledMarkdown';
+import { SectionContext } from '~/context';
+import data from 'data/casestudies.json';
+import Card from '~/components/Card';
+export async function getStaticProps() {
```
Comment (benjagm) [inline pages/overview/casestudies/index.page.tsx:11]: Let's not use a md file here. Let's add all the content here.

File: pages/overview/casestudies/index.page.tsx [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+export default function ContentExample({ blocks }: { blocks: any[] }) {
+  const newTitle = 'Casestudies';
+
+  return (
+    <SectionContext.Provider value='docs'>
+      <Head>
+        <title>{newTitle}</title>
+      </Head>
+      <Headline1>{newTitle}</Headline1>
+      <StyledMarkdown markdown={blocks[0]} />
```
Comment (benjagm) [inline pages/overview/casestudies/index.page.tsx:33]: It will be great to have a title and an intro paragraph. You can use the typical lorem ipsum.

**Review by benjagm — CHANGES_REQUESTED** [review]
Amazing progress. We are really close.

Left some comments. Have you done a rebase to bring the changes from the new parent branch?

File: pages/overview/casestudies/index.page.tsx (hunk 10 above)
Comment (TamannaVerma99) [reply pages/overview/casestudies/index.page.tsx:33]: Thanks for the feedback :) 
I will soon implement all the points mentioned above.

File: data/casestudies.json (hunk 4 above)
Comment (TamannaVerma99) [reply data/casestudies.json:1]: Hey @benjagm 
where do we want to direct the link we are adding in each casestudy card? According to card's implementation the link I am adding is shown as read more.
e.g if I have added homepage url in link prop of card it is showing read more and on clicking read more i am directed to home page.

File: components/Sidebar.tsx (hunk 5 above)
Comment (TamannaVerma99) [reply components/Sidebar.tsx:187]: This change is not needed ,sorry for the confusion from my side.. I m resolving it.

File: pages/overview/casestudies/_index.md (hunk 8 above)
Comment (TamannaVerma99) [reply pages/overview/casestudies/_index.md:1]: Working on it :)

File: data/casestudies.json (hunk 4 above)
Comment (benjagm) [reply data/casestudies.json:1]: > where do we want to direct the link we are adding in each casestudy card?

The url will be the case study blog post.

Comment (TamannaVerma99) [reply data/casestudies.json:1]: Okay! Got it

Comment (TamannaVerma99) [reply data/casestudies.json:1]: Hey @benjagm I have some doubts regarding data set
What are we going to add at the place of logo and are we going to add complete title of blog post if yes then it will be too large because in the implementation of card title it is made by using h3 so how should I proceed with it.

Comment (benjagm) [reply data/casestudies.json:1]: We need to find the company logos and add them to the folder: https://github.com/json-schema-org/website/tree/main/public/img/logos

Comment (benjagm) [reply data/casestudies.json:1]: Why changing the size of the card? Try to use this one and if you need to make some adjustments do it but just for the card features that you need. Remember that the card is now going to be used in 3 different pages. 

For example you can change the logo size or fix a bug but not the whole card component without reason. Right?

Comment (TamannaVerma99) [reply data/casestudies.json:1]: Sorry for the confusion @benjagm 
I am not asking to change the card size but the bug is related to title if I directly put title in this component it will look like this

cr

It is  because title is written with h3 size in card component.
I will definitely not change the card component directly but if there is no other way I will just make a copy of this component and make changes to that one .

Comment (benjagm) [reply data/casestudies.json:1]: In this case feel free to modify it

**Comment by TamannaVerma99** [comment]
Hey @benjagm .
Because of the issues caused in this PR since the branch was based on main instead of web-release-3.0, I have now created a new PR #540 with all the recommended changes.  I would suggest closing this PR hereafter, and continuing our discussions on the other one. Thanks!

**Review by benjagm — APPROVED** [review]
Amazing job!!! This is ready to go to the dev branch. I pushed some code to make some adjustments with the logos but really great work.

Congrats!!!!

**Comment by TamannaVerma99** [comment]
> Amazing job!!! This is ready to go to the dev branch. I pushed some code to make some adjustments with the logos but really great work.
> 
> Congrats!!!!

Thanks:)
